src/multiplexed-session.ts

```diff
@@ -0,0 +1,320 @@
+  constructor(
+    database: Database,
+    multiplexedSessionOptions?: MultiplexedSessionOptions
+  ) {
+    super();
+    this.database = database;
+    this.options = Object.assign({}, DEFAULTS, multiplexedSessionOptions);
+    this.options.databaseRole = this.options.databaseRole
+      ? this.options.databaseRole
+      : database.databaseRole;
+    this._muxSession = {
+      multiplexedSession: null,
+    };
+    this._acquires = new PQueue({
+      concurrency: 1,
+    });
+  }
+
+  /**
+   * Creates a new multiplexed session and manages its maintenance.
+   *
+   * This method initiates the session creation process by calling the `_createSession` method, which returns a Promise.
+   */
+  createSession(): void {
+    this._createSession()
+      .then(() => {
+        this._maintain();
+      })
+      .catch(err => {
+        if (
+          isDatabaseNotFoundError(err) ||
+          isInstanceNotFoundError(err) ||
+          isCreateSessionPermissionError(err) ||
+          isDefaultCredentialsNotSetError(err) ||
+          isProjectIdNotSetInEnvironmentError(err)
+        ) {
+          return;
+        }
+        this.emit('error', err);
+      });
+  }
+
+  /**
+   * Creates a new multiplexed session.
+   *
+   * This method sends a request to the database to create a new session with multiplexing enabled.
+   * The response from the database would be an array, the first value of the array will be containing the multiplexed session.
+   *
+   * @returns {Promise<void>} A Promise that resolves when the session has been successfully created and assigned, an event
+   * `mux-session-available` will be emitted to signal that the session is ready.
+   *
+   * @private
+   */
+  async _createSession(): Promise<void> {
+    try {
+      const [createSessionResponse] = await this.database.createSession({
+        multiplexed: true,
+      });
+      this._muxSession.multiplexedSession = createSessionResponse;
+      this.emit('mux-session-available');
+    } catch (e) {
+      this.emit('error', e);
+    }
+  }
+
+  /**
+   * Maintains the multiplexed session by periodically refreshing it.
+   *
+   * This method sets up a periodic refresh interval for maintaining the session. The interval duration
+   * is determined by the @param refreshRate option, which is provided in minutes.
+   * The default value is 7 days.
+   *
+   * @returns {void} This method does not return any value.
+   *
+   * @throws {Error} If the `refreshRate` option is not defined, this method could throw an error.
+   */
+  _maintain(): void {
+    const refreshRate = this.options.refreshRate! * 24 * 60 * 60000;
+    this._refreshHandle = setInterval(() => this._refresh(), refreshRate);
+    this._refreshHandle.unref();
+  }
+
+  /**
+   * Creates a transaction for a session.
+   *
+   * @private
+   *
+   * @param {Session} session The session object.
+   * @param {object} options The transaction options.
+   */
+  _prepareTransaction(session: Session | null): void {
```

Comment:
This is again a duplicate method ?

Reply:
Since, this method is part of SessionPool class and is private; to use it in this MultiplexedSession class we need to create an object of the SessionPool class. Hence, I will move this method in the GetSession class later in the other PR. So, that from that common class both the SessionPool and MultiplexedSession can use it